src/posix/platform/logging.cpp

```diff
@@ -69,7 +75,46 @@ OT_TOOL_WEAK void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const
     }
 
     va_start(args, aFormat);
+#if OPENTHREAD_POSIX_CONFIG_LOG_OUTPUT_STDOUT_PRINTF
+    const uint16_t kBufferSize = 4096;
+    char           buffer[kBufferSize];
+    OT_UNUSED_VARIABLE(aLogLevel);
+    if ((vsnprintf(buffer, sizeof(buffer), aFormat, args) > 0))
+    {
+        struct timeval tv;
+        time_t         nowtime;
+        struct tm     *nowtm;
+        char           tmbuf[64], buf[64];
+        gettimeofday(&tv, nullptr);
+        nowtime = tv.tv_sec;
+        nowtm   = localtime(&nowtime);
+        strftime(tmbuf, sizeof tmbuf, "%Y-%m-%d %H:%M:%S", nowtm);
+
+#if defined(__APPLE__)
+        int ret = snprintf(buf, sizeof buf, "%s.%06d", tmbuf, tv.tv_usec);
+#else
+        int ret = snprintf(buf, sizeof buf, "%s.%06ld", tmbuf, tv.tv_usec);
```

Comment:
Same here, `ret` should be defined at the top of scope (after `{`)

Reply:
Updated